### ISSUE TITLE
long path doesn't match the policy

### ISSUE THREAD

**Issue by pa-federici** [kubearmor/KubeArmor#1460]
**General Information**

- Environment description  eks 1.28
- Kernel version Linux bash 5.10.192-183.736.amzn2.x86_64  UTC 2023 x86_64 x86_64 x86_64 GNU/Linux
- Orchestration system version in use v1.28.1
- Target issue riproduced on different containers, including on standard images from docker hub like ubuntu:23.04
- Tested with `kubearmor/kubearmor` and `kubearmor/kubearmor-init` image versions `v1.0.1` and `v0.11`

**To Reproduce**

1.  Create a pod with the image ubuntu:23.04
2.  Create a directory called "test" in the root of the container, and a sub directory with a 250  characters name
3.  Set the "DefaultPosture" to Block
4.  Create  a Kubearmor policy that match the container, adding all the root container directory, setting action parameter to "Allow" and recursive parameterto "true" for all the directory ( including "test" folder).
5. Apply the policy
6. Execute commands inside the subdirectory created in the point 2.
Es:
    -  `ls`
    - `echo "hello word" >  file.txt`
    - `touch file.txt`

**Expected behavior**

We expect that the container match with the policy created that allows operations on that path, and not the DefaultPosture that blocks them.

**Logs**
```json

{
    "Timestamp":1697038422,
    "UpdatedTime":"2023-10-11T15:33:42.211351Z","ClusterName":"stm-vespucci-dev-eks-cluster",
    "HostName":"ip-192-168-1-72",
    "NamespaceName":"test-kubearmor",
    "Owner":{"Ref":"Pod","Name":"bash",
    "Namespace":"test-kubearmor"},
    "PodName":"bash",
    "Labels":"test-kubearmor=true","ContainerID":"9976a024b018cbacac251e7fa072f4f777c28ead2fc8167e8567db438ea7f126","ContainerName":"bash",
    "ContainerImage":"docker.io/library/ubuntu:23.04@sha256:f1090cfa89ab321a6d670e79652f61593502591f2fc7452fb0b7c6da575729c4","HostPPID":1026888,"HostPID":1026980,"PID":1,
    "ParentProcessName":"/usr/bin/containerd-shim-runc-v2",
    "ProcessName":"/usr/bin/bash",
    "PolicyName":"DefaultPosture",
    "Type":"MatchedPolicy",
    "Source":"/usr/bin/bash",
    "Operation":"File",
    "Resource":"test.txt",
    "Data":"syscall=SYS_OPENAT fd=-100 flags=O_WRONLY|O_CREAT|O_TRUNC","Enforcer":"eBPF Monitor",
    "Action":"Block",
    "Result":"Permission denied"
}

```
```json

{  
    "Timestamp":1697038960,
    "UpdatedTime":"2023-10-11T15:42:40.878325Z","ClusterName":"stm-vespucci-dev-eks-cluster",
    "HostName":"ip-192-168-1-72",
    "NamespaceName":"test-kubearmor",
    "Owner":{"Ref":"Pod","Name":"bash","Namespace":"test-kubearmor"},"PodName":"bash","Labels":"test-kubearmor=true","ContainerID":"9976a024b018cbacac251e7fa072f4f777c28ead2fc8167e8567db438ea7f126","ContainerName":"bash",
    "ContainerImage":"docker.io/library/ubuntu:23.04@sha256:f1090cfa89ab321a6d670e79652f61593502591f2fc7452fb0b7c6da575729c4","HostPPID":1026888,
    "HostPID":1026980,
    "PID":1,
    "ParentProcessName":"/usr/bin/containerd-shim-runc-v2",
    "ProcessName":"/usr/bin/bash",
    "PolicyName":"DefaultPosture",
    "Type":"MatchedPolicy",
    "Source":"/usr/bin/bash",
    "Operation":"File","Resource":"LIORFU4KPSIWZNZQMLLYFBDHNDCWHSOJFTAI2P7SGONTG2ECPX77L34U7IJXR4RGRAM3365GRAQKNZDVQVCHJGNBUUAN4SRAXSQ46RE55HI2SPECO6FT5ZXBTMOPY5RXPGTTZOA4FFBW5JY5UGN2VIIN44GDXRTH37UB5C6WH2FWBOWWCKBLI6ICTXAX2SUCBJCM364SJ5JVD3YT7FVDZLHRKB63XZ7JWGZTYNUDFSPFCVPVUBZBM3XWEZ/test.txt",
    "Data":"syscall=SYS_OPENAT fd=-100 flags=O_WRONLY|O_CREAT|O_TRUNC","Enforcer":"eBPF Monitor",
    "Action":"Block",
    "Result":"Permission denied"
}
```

**Comment by ernspa94** [comment]
Hi @daemon1024 , i have the same problem. Any update on this?


Thank you

**Comment by redefik** [comment]
+1

**Comment by daemon1024** [comment]
Hey Folks, We have a size constraint of 256 string length of path.

For some reason we had the constraint set to 64 but it's been updated to 256 now in #1484.

We will make some updates to our code such that directory rules are still enforceable. And warn the users about how to not include paths greater than 256 length in policies.

Look forward to updates in v1.1.0. For a quick fix once #1484 is merged you can find the expected behaviour upto 256 path length.

**Comment by daemon1024** [comment]
Folks, please verify with latest kubearmor
you can use `karmor install --tag=latest` 
Or if using helm update tag from stable to latest in the configuration file.

Noting policy specification still would only support maximum of 256 characters, but enforcement on directory is fixed. Thanks.
